# Profitable Apps To Consider Building By Appstore

## Introduction

### Goal
The goal is to find trends on apps that are sucessful to guide product development.

This analysis will make sure resources are being utilized correctly and set up for success as much as possible before launch.

### Items To Consider
Most revenue of the revenue we want will come from add and if the data supports our assumptions the app should be free to attract as many users. As more exposure, more gains. App should be in the English language.

Google Play vs App Store. If one store can lead to more revenue we can focus on lauching an app in one store first before another. If the relative impact is the samee, and we have enough resources we can have two project teams working towards the same launch day.

Links to data sets:

Google Play, [here](https://dq-content.s3.amazonaws.com/350/googleplaystore.csv)

iOS Appstore, [here](https://dq-content.s3.amazonaws.com/350/AppleStore.csv)

In [1]:
#Opening Data Sets#

from csv import reader

opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

In [2]:
def explore_data(dataset, start, end, rows_and_columns = False):
    dataset_slice = dataset[start:end]
    for row in dataset_slice:
        print(row)
        print('\n')

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns', len(dataset[0]))

print(android_header)
print('\n')
explore_data(android,0,3,True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns 13


In [3]:
print(ios_header)
print('\n')
explore_data(ios,0,3,True)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


Number of rows: 7197
Number of columns 16


# Before Going Further

## Google Play
Number of rows: 10841
Number of columns 13

Based on a cursory review of the headers. The columns that we most likely need for our analysis are:
- App
- Category
- Installs
- Reviews
- Type
- Price
- Genres


## iOS AppStore
Number of rows: 7197
Number of columns 16

Similar to Google Play Data Set. The columns that we most likely need for our analysis are:
- Track Name
- Currency
- Price
- Rating_count_tot
- Rating_count_ver
- Prime_genre

Descriptions to column descriptions for each data set can be found here:
[Google Play](https://www.kaggle.com/lava18/google-play-store-apps)
[iOS AppStore](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps)

# Data Cleansing

## Missing Data in Google Play Store
Based on some discussions on Kaggle (links above) some users detailed that there was a row with missing column that was causing issues. The row was removed as to not cause issues later.

## iOS AppStore
Based on the Kaggle discussions and some testing there does not appear to be any missing data that could cause a shift.

In [4]:
print(android[10472])
print('\n')
print(android_header)
print('\n')
print(len(android))
del android[10472]
print(len(android))

['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


10841
10840


# Data Cleansing, Continued

## Checking for Dupes in Google Play Store
First we will create two list, one of unique values and one with the duplicate apps. We will be relying on the name of the app as the Google Play Store does not have unique ids we can rely on to reliably clean the data.

Once we identify the dupes, we will remove all the ones with the lowest amount of reviews as the Google Play Store does not capture updated date/times and we are assuming that the more reviews the more recent it is (although we would have to verify removing reviews policy from Google Play Store to make sure this logic is sound).

In [5]:
duplicate_apps = []
unique_apps = []

for app in android:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print('Number of duplicate apps:', len(duplicate_apps))
print('\n')
print('Expected length after deduping:', len(android) - len(duplicate_apps))
print('\n')
print('Examples of duplicate apps:', duplicate_apps[:15])

Number of duplicate apps: 1181


Expected length after deduping: 9659


Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


## Removing Dupes

We will need to create a for loop where it will check for an app name in a dictionary and evaluate if the reviews is less than the curent iteration, if less than it will swap to the higher review number. If app name is not present it will add to the dictionary along with its reviews number.

After a quick check that our dictionary value contains the expected length. We will leverage that dictionary to create a clean data list of the Google Play Store data set.

In [6]:
reviews_max = {}

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
    elif name not in reviews_max:
        reviews_max[name] = n_reviews
print(len(reviews_max))

android_clean = []
already_added = []

for app in android:
    name = app[0]
    n_reviews = float(app[3])
    
    if (reviews_max[name] == n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

9659


## Removing Dupes, Continued

We will once again use our previously created function to check out last time that we end up with our expected total before continuing.

In [7]:
explore_data(android_clean, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9659
Number of columns 13


## Checking for Dupes in iOS AppStore

We will follow a similar procedure as we did for the Google Play Store data set but instead we will use the id and not track_name column.

In [8]:
duplicate_apps = []
unique_apps = []

for app in ios:
    name = app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
print('Number of duplicate apps:', len(duplicate_apps))

Number of duplicate apps: 0


## Data Cleansing, Continued

As you can see from above the iOS App Store data does not contain any duplicates but our job of cleaning the data is not complete.

## Removal of Non-English Apps

Based on our assumptions, we want to remove any apps that are not in English language. This can be checked by using Unicode to remove any non Latin based charecters (English Alphabet), numbers composed of digits, punctuation marks and other symbols.

Based on [ASCII](https://www.ascii-code.com/), we can determine that any unicode number equal to or less than 127 is used in the English Language.

In [9]:
def is_english(string):
    for charecter in string:
        if ord(charecter) > 127:
            return False
    return True

print(is_english('Instagram'))
print(is_english('WeChat, 微信'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
False
False
False
False


## Removal of Non-English Apps, Continued

As you can see from the example above of us testing our function to make sure we aren't overly removing app names. App names with charecters are being removed even though they are in English. We will need to refine our function a bit more so that we are not removing to many apps and retest our examples to make sure it's working properly.

In [10]:
def is_english(string):
    non_ascii = 0
    
    for charecter in string:
        if ord(charecter) > 127:
            non_ascii += 1
    if non_ascii > 3:
        return False
    else:
        return True

print(is_english('Instagram'))
print(is_english('WeChat, 微信'))
print(is_english('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(is_english('Docs To Go™ Free Office Suite'))
print(is_english('Instachat 😜'))

True
True
False
True
True


## Removal of Non-English Apps, Continued

As you can see from above, our function isn't fully working how we want to as WeChat a very common app is still being considered an English app. We can come back an optimize at a later date.

For now we will use our function on both data sets below.

In [11]:
android_english = []
ios_english = []

for app in android_clean:
    name = app[0]
    if is_english(name):
        android_english.append(app)
        
for app in ios:
    name = app[1]
    if is_english(name):
        ios_english.append(app)
        
explore_data(android_english, 0, 3, True)
print('\n')
explore_data(ios_english, 0, 3, True)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns 13


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', '

## Finding Free Apps

After running our is_english function we can see that bothd data sets have gotten smaller. At this point we have removed inaccurate data, removed duplicates and removed non-english apps.

Our goal is to create a free app as we are assuming that will give us the most exposure and eyeballs for our ads. We need to create a function to find the free apps so we can continue our analysis.

In [12]:
android_final = []
ios_final = []

for app in android_english:
    price = app[7]
    if price == '0':
        android_final.append(app)
for app in ios_english:
    price = app[4]
    if price == '0.0':
        ios_final.append(app)

print(len(android_final))
print(len(ios_final))        

8864
3222


# Analysis: Most Common Apps by Genre

## Data Cleansing Recap:
Now that we have completed our data cleansing (items completed below):
- Removed inaccurate data
- Removed duplicate app entries
- Removed non-English apps
- Isolated the free apps

We are left with 8864 apps in the Google Play Store data set and 3222 in the iOS AppStore data set to do our analysis.

Our strategy to keep cost and resources at minimum is to create an Android app add it to the Google Play Store and if we get a good response from users then we can further develop it. If the app becomes profitable after sometime on the Google Play Store, then we can begin building the app for iOS. Since our strategy is to be on both stores we will need an app that can be sucessful on both apps stores.

We can achieve this with a frequency table to analyze the genres of most common types of apps for both app stores. Based on the header names and documentation for the Google Play Store, we can use column one and nine and for the iOS store we can use column eleven for analysis.

In [13]:
def freq_table(dataset,index):
    table = {}
    total = 0
    
    for row in dataset:
        total +=1
        value = row[index]
        if value in table:
            table[value] +=1
        else:
            table[value] = 1
    
    tables_percentages = {}
    for key in table:
        percentage = (table[key] / total) * 100
        tables_percentages[key] = percentage
    return tables_percentages

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)
         
    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

In [14]:
display_table(ios_final, 11)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


## iOS App Store Analysis

We can see based on the frequency of apps the top five categories for free apps consist of games, entertainment, photo/video, education and social networking. Now to analyze the Googly Play Store dataset.

In [15]:
display_table(android_final,1)
print('\n')
display_table(android_final, 9)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

## Google Play Store Analysis

Based on running the frequency function on both columns we can see that the generes column is more granular and gives us more insight into what type of app it is compared to the Category which seems more generic.

The top five most frequent types of apps that are free are tools, entertainment, education, business and productivity.

## Looking at both

We can see that most iOS apps are centered around entertainment (games,entertainment, social networking). While the data/apps in the Google Play Store appears to be more diverse while entertainment is the second most common type of app the rest of the top five are (tools, education, business, productivity).

Currently we are looking at the most common type of app but that does not mean they have all have large amounts of users, if they are more common then the pool of potential users can be split among more apps. Finding a way to stand out from the the most common may be difficult depending for which type of app we want to build for.

To be able to give direction, I think we will need to find a way to see the average of how many users/downloads each genre of app has.

In [16]:
genres_ios = freq_table(ios_final, 11)

for genre in genres_ios:
    total = 0
    len_genre = 0
    for app in ios_final:
        genre_app = app[11]
        if genre_app == genre:
            n_ratings = float(app[5])
            total += n_ratings
            len_genre += 1
    avg_n_ratings = total / len_genre
    print(genre, ':', avg_n_ratings)

Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


## Downloads, iOS App Store
Based on the average downloads above, we can start getting a better picture of which direction we should go at least from the perspective of the iOS App Store. Our current data above is not sorted like before but very quickly we can see social networking, navigation and reference are high. All three are averaging about 70k plus reviews which in this case we are using in lieu of downloads since iOS data set does not have that column for us.

We should further analyze those three types of apps to make sure that there are no big outliers which I suspect there are since we are using averages to do our analysis at this stage.

In [17]:
for app in ios_final:
    if app[11] == 'Navigation':
        print(app[1], ':', app[5])
print('\n')
for app in ios_final:
    if app[11] == 'Social Networking':
        print(app[1], ':', app[5])
print('\n')
for app in ios_final:
    if app[11] == 'Reference':
        print(app[1], ':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


Facebook : 2974676
Pinterest : 1061624
Skype for iPhone : 373519
Messenger : 351466
Tumblr : 334293
WhatsApp Messenger : 287589
Kik : 260965
ooVoo – Free Video Call, Text and Voice : 177501
TextNow - Unlimited Text + Calls : 164963
Viber Messenger – Text & Call : 164249
Followers - Social Analytics For Instagram : 112778
MeetMe - Chat and Meet New People : 97072
We Heart It - Fashion, wallpapers, quotes, tattoos : 90414
InsTrack for Instagram - Analytics Plus More : 85535
Tango - Free Video Call, Voice and Chat : 75412
LinkedIn : 71856
Match™ - #1 Dating App. : 60659
Skype for iPad : 60163
POF - Best Dating App for Conversations : 52642
Timehop : 49510
Find My Family, Friends & iPhone - Life360 Locator : 43877
Whisper - Share, Express, 

## Downloads, iOS App Store Continued

Our suspicions were right. There are some major apps that are making the reviews average be higer. But the reference category seems promising based on the diversity of the types of apps. If we could find a book or some sort of knowledge base that people need to reference and add more features we could have an interesting app to attract new users.

Now we need to apply this same analysis to the Google Play Store data set but the only big difference will be that the Google Play Store data set does have a downloads (column 5, titled "installs") category which will help us make a better decision instead of infering through the reviews column.

In [18]:
display_table(android_final,5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


## Downloads, Google Play Store

One things we notice very quickly with the installs column in the Google Play data set is that the number are not precise but instead are wide ranges. For example at the top end we don't know if an app has 100k installs or 300k installs all we know is it falls between 100k to 1m which is a pretty big range.

Luckily for us, we are looking to make an informed decision and not looking at precise numbers. To do our analyses we will need to convert that column to integers with the float function and we will just assume 100k+ is just equal to 100k.

In [19]:
categories_android = freq_table(android_final,1)

for category in categories_android:
    total = 0
    len_category = 0
    for app in android_final:
        category_app = app[1]
        if category_app == category:
            n_installs = app[5]
            n_installs = n_installs.replace(',','')
            n_installs = n_installs.replace('+','')
            total += float(n_installs)
            len_category += 1
    avg_n_installs = total / len_category
    print(category,':', avg_n_installs)

ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

## Downloads, Google Play Store, Continued

Based on the data above our biggest  categories are:
* COMMUNICATION 
* VIDEO_PLAYERS 
* SOCIAL 
* PHOTOGRAPHY 
* PRODUCTIVITY 
* GAME 
* TRAVEL_AND_LOCAL 
* ENTERTAINMENT 
* TOOLS 
* NEWS_AND_MAGAZINES 
* BOOKS_AND_REFERENCE

But similar to the iOS App Store data we may have some larger players skewing the average. Which we will confirm below.

In [20]:
for app in android_final:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])
print('\n')
for app in android_final:
    if app[1] == 'PRODUCTIVITY' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])
print('\n')
for app in android_final:
    if app[1] == 'TOOLS' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

## Downloads, Google Play Store, Continued

As we suspected, ignoring the entertainment, games section if we take a look at some of of ther other categories we can see some big apps skewing the data. If we recall the Books and Reference category in iOS App Store looked promising so we should analyze that more closely to see if it is also a good match for us in the Google Play Store data set.

In [21]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                      or app[5] == '500,000,000+'
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


## Downloads, Google Play Store, Continued

We can see that the list for this category is substantially smaller than the other categories. Only 5 big apps and if we look for smaller installs we should see a bigger list that is more diverse which is what we want.

In [22]:
for app in android_final:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                            or app[5] == '5,000,000+'
                                            or app[5] == '10,000,000+'
                                            or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

# Conclusion

Across iOS and Android app store data we can see that If we could find a book or some sort of knowledge base that people need to reference and add more features we could have an interesting app to attract new users.